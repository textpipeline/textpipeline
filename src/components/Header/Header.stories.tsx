import { storiesOf } from '@storybook/react';
import React from 'react';

import Header from './Header';

storiesOf('Header', module).add('default', () => <Header siteName="Test Site" />);
