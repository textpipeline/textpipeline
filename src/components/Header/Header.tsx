import Link from '@material-ui/core/Link';
import Typography from '@material-ui/core/Typography';
import React from 'react';

export interface HeaderProps {
  readonly siteName: string;
}

const Header: React.FC<HeaderProps> = ({ siteName }) => (
  <Typography variant="h4" component="h1" noWrap align="center">
    <Link href="/" underline="none">
      {siteName}
    </Link>
  </Typography>
);

export default Header;
