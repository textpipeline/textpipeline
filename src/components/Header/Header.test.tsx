import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Header from './Header';

describe('Header', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<Header siteName="Test Site" />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
