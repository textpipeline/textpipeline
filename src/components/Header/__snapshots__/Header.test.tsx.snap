// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  align="center"
  component="h1"
  noWrap={true}
  variant="h4"
>
  <WithStyles(ForwardRef(Link))
    href="/"
    underline="none"
  >
    Test Site
  </WithStyles(ForwardRef(Link))>
</WithStyles(ForwardRef(Typography))>
`;
